From fdacf58b965ad28253ea4da4215b1e9e11b2c92f Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Fri, 28 Dec 2012 13:54:03 +0800
Subject: [PATCH 379/449] ENGR00238439 ASRC TEST: add tail for output buffer
 length

ASRC can't promise the length of output buffer length when the input
buffer length and params is fixed. ASRC driver would cut off part of
data which is larger than the expected output buffer length. However,
the cut operation would cause noise.

In this patch, add some tail for output buffer length to prevent noise.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 test/mxc_asrc_test/mxc_asrc_test.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/mxc_asrc_test/mxc_asrc_test.c b/test/mxc_asrc_test/mxc_asrc_test.c
index f18bdaf..2ba33a4 100644
--- a/test/mxc_asrc_test/mxc_asrc_test.c
+++ b/test/mxc_asrc_test/mxc_asrc_test.c
@@ -191,6 +191,7 @@ int play_file(int fd_asrc, struct audio_info_s *info)
 	output_dma_size =
 	    asrc_get_output_buffer_size(DMA_BUF_SIZE, info->sample_rate,
 					info->output_sample_rate);
+	tail = info->channel * 4 * 2;
 
 	convert_flag = 1;
 	memset(input_null, 0, DMA_BUF_SIZE);
@@ -212,7 +213,7 @@ int play_file(int fd_asrc, struct audio_info_s *info)
 			buf_info.input_buffer_length = DMA_BUF_SIZE;
 		}
 
-		buf_info.output_buffer_length = output_dma_size;
+		buf_info.output_buffer_length = output_dma_size + tail;
 		buf_info.output_buffer_vaddr = output_p;
 		if ((err = ioctl(fd_asrc, ASRC_CONVERT, &buf_info)) < 0)
 			goto error;
-- 
1.8.0

